src/LocalMesh.C

```diff
@@ -251,8 +251,8 @@ void LocalMesh::InitSolution(double *flovar, int nfields)
   q=gpu::allocate_on_device<double>(sizeof(double)*(ncells+nhalo)*nfields);
   qn=gpu::allocate_on_device<double>(sizeof(double)*(ncells+nhalo)*nfields);
   qnn=gpu::allocate_on_device<double>(sizeof(double)*(ncells+nhalo)*nfields);
-  res_d=gpu::allocate_on_device<double>(sizeof(double)*(ncells+nhalo)*nfields);
-  dq_d=gpu::allocate_on_device<double>(sizeof(double)*(ncells+nhalo)*nfields);
+  res_d=gpu::allocate_host_pinned<double>(sizeof(double)*(ncells+nhalo)*nfields);
```

Comment:
Why allocate res_d and dq_d as host pinned ?  I get that for the qbuf, but won't this be inefficient if we host pin the residual and dq, the main arrays that get filled in by GPU kernels ?

Reply:
I believe I made this change b/c both `res_d` and `dq_d` are passed to the `UpdateFringes`. Yes, I agree this is definitely not ideal -- I am still fiddling with it.